Updates written at sprint after plonetagung in Munich

pbauer opened the pull request.

pbauer committed:
fix example code about Products.PDBDebugMode

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -89,14 +89,22 @@ Then the source of the add-on package will be checked out into the ``src`` folde
 
-    You can also add Products.PDBDebugMode and plone.reload to your development-tools:
+    You can also add ``Products.PDBDebugMode`` and ``plone.reload`` to your development-tools. For ``Products.PDBDebugMode`` you currently need a source-checkout though.
 
```
Please follow our style guide.
One sentence per line.
https://docs.plone.org/about/contributing/documentation_styleguide.html

Thanks!

stevepiercy replied:
Fixed in 9f40573.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -107,13 +115,13 @@ Then the source of the add-on package will be checked out into the ``src`` folde
 Now everything is prepared to work on the migration of the package.
 
-However, it is a good idea to now try
+However, for small or simple packages it is a good idea to now try
```
Please follow our style guide!
https://docs.plone.org/about/contributing/word_choice.html

Thanks!

stevepiercy replied:
Fixed in 9f40573.

pbauer committed:
update branch for PDBDebugMode

staeff committed:
Add six to requirements of package

Mention that six needs to be added to install_requires in setup.py

staeff committed:
Add Jenkins testing info

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -148,6 +148,8 @@ The easiest way is to use ``isort``.
 Check the `Python style guide for Plone <https://docs.plone.org/develop/styleguide/python.html#grouping-and-sorting>`_
 for information about the order of imports and an example configuration for ``isort``.
 
+If ``six`` is used in the code,
```
I would like to suggest to add a here a link to the package or its documentation, for example: https://six.readthedocs.io/

Doing so is handy because it gives people more background info.
Not everyone knows ``six``

Thanks!

stevepiercy replied:
The first instance of six in this document is already hyperlinked.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -306,6 +306,14 @@ With some luck, there are not too many issues with the code left at this point.
 It you are unlucky you have to fix Doctests. These should be changed so that Python 3 is the default. (e.g. text is represented as ``'foo'`` and not ``u'foo'`` and bytes are represented as ``b'bar'`` and not as ``'bar'``). Search for examples of ``Py23DocChecker`` in Plone's packages to find a pattern which allows updated doctests to pass in Python 2.
 
 
+Test your code against buildout.coredev on Jenkins (https://jenkins.plone.org/view/Add-ons/). 
```
Can we make that as an nice link, meaning that "buildout.coredev" is the link text ?

Thanks!

stevepiercy replied:
Fixed in 9f40573.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -306,6 +306,14 @@ With some luck, there are not too many issues with the code left at this point.
+Fill the fields ADDON_URL and ADDON_BRANCH with your repo url and the branch name 
+(python3 if you followed this instructions). 
+Start the build with the 'Build' button.
 
```
Is the intention here that this is a list, like a bullet list?

stevepiercy replied:
Fixed in f41a0cf.

staeff committed:
Info about StringIO/BytesIO imports

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -361,8 +361,16 @@ Try to modify the code in such a way that when dropping support for py2 you will
 
+``python-modernize`` also does not touch the import statement ``from StringIO import StringIO`` even though
+this is only works in Python 2. 
+You have the check whether you are dealing textual or binary data and use the according import statement from ``six`` (https://pythonhosted.org/six/#six.StringIO)
```
What about:
You have the check whether you are dealing textual or binary data and use the according import statement from `six <https://pythonhosted.org/six/#six.StringIO>`_

stevepiercy replied:
Fixed in 9f40573.

pbauer committed:
remove obsolete checkout

ksuess committed:
Typo

stevepiercy committed:
Grammar, syntax, style fixes up through "9 Frequent Issues".
- Hyperlink first instance of packages to their docs.
- One sentence per line.
- "precompiler" not "precompile" for package name.

stevepiercy committed:
Comment out invalid Python syntax for highlighting.

stevepiercy committed:
Change paragraph to a bulleted list of steps.

stevepiercy reviewed (commented):
I've run through this document up through "9 Frequent Issues", stopping at "Database Migration". I'd appreciate a review of my changes, so that I don't self-approve this PR.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -15,27 +15,26 @@ Principles
+* Open a ticket with the title "Add support for Python 3".
 * Create a new branch named ``python3``.
 
 Until Plone 5.2 is released, you can use the coredev buildout setup.
```
This are supposed to be the docs for Plone 5.2, meaning this line 40 should be removed.

stevepiercy replied:
If it is removed, then the subsequent sentence will not make sense. Perhaps just remove the phrase "Until Plone 5.2 is released, " and capitalize "you"?

svx replied:
Maybe I am not understanding it?
If I look at the PR request, this is a request to merge it into the 5.2 branch.

Besides that, the branch is public on GitHub there is no other release of this branch.
AFAIK we will release it to docs.plone.org during the release of Plone 5.2.

Because of this, I am wondering why we need this line in the docs.
For me, this would make only sense if these docs would be merged into current 'master' (HTML) on docs.plone.org.

Do I not getting something?

stevepiercy replied:
I agree about v5.2.

However, lines 40-53 all refer to using "coredev buildout setup". Removing all of line 40 will make lines 41-53 difficult to understand.

svx replied:
OK, just read the whole part in a different browser tab.
Yes, that is indeed a problem!
The question for me is now if that part should be there like this at all?
The current part makes complete sense if you work on this before 5.2 is released, but after the release, this is (as I understand it0 obsolete (lite it is written now) in the 5.2 docs.
So we need a solution where we have now the current part, but after the release, that needs to change.
What is with the 'dirty' idea to keep it like it is for now and open a ticket (with a reminder) to change that after the release?
I can put it on my release checklist for example.

stevepiercy replied:
We could put a `..warning::` preceding the section:

```rst
..warning::

    The following section is valid until the final release of Plone 5.2.
    Upon the final release of Plone 5.2, [something else will take its place].
```

We could also use `..todo::` or comment `..` in reST.

svx replied:
Yes! I would go with the warning, awesome idea!

stevepiercy replied:
Fixed in f1a5785. Anything else need to be noted, new issue, added to checklist?

svx replied:
Added to checklist! Will polish the checklist a bit and will add it to GitHub sometime during next week.
Guess it is handy that more people than two, know the steps and how to fix common issues during our currently somewhat hacky and old way.
Well from Plone 6 on that will be way better.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -89,31 +88,43 @@ Then the source of the add-on package will be checked out into the ``src`` folde
-    You can also add Products.PDBDebugMode and plone.reload to your development-tools:
+    You can also add some development-tools like ``Products.PDBDebugMode`` and ``plone.reload`` to your buildout.
+    For some you currently need a source checkout though.
 
```
Add a link to docs about source checkout

stevepiercy replied:
What is the link? I don't know the correct one.

svx replied:
This is a good question, I would go with https://training.plone.org/5/mastering-plone/buildout_1.html#buildout-i
Even it is not perfect it gives useful info.
Longtime Plone devs may not need it, but if you are new, this may help you.

stevepiercy replied:
I'm also unclear about whether "For some" refers to buildouts or people doing buildouts. Can you clarify?

svx replied:
As I read and understand it that "For some" is meaning "For some development tools"
But this is only my guess :)

stevepiercy replied:
Added link for source checkout in 0acc5b7

stevepiercy replied:
Improved wording slightly in 8ef23c8. It's still klunky.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -89,31 +88,43 @@ Then the source of the add-on package will be checked out into the ``src`` folde
 
-However, it is a good idea to now try
+However, for small packages or packages that have few dependencies, it is a good idea to try starting your instance now.
 
```
Can we remove/change the word "however" ?
It is not needed here, cause it adds no extra value and some people could read that as an insult and this is not good style in terms of tone.

stevepiercy replied:
Done in 573b21c

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -125,29 +136,26 @@ There are some problems that ``python-modernize`` can not fix on its own.
+Install ``isort`` into your Python 3 environment with ``pip``.
 
 .. code-block:: shell
 
```
Why are the links removed, do we link to these packages already somewhere else?
Linking to packages/docs is good practice, not everyone may know or use them already.

stevepiercy replied:
See line 144, the first time `isort` is mentioned.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -184,21 +191,22 @@ You can use ``isort`` to fix the order of imports:
+You can make use of ``plone.recipe.precompiler`` to identify syntax errors quickly.
 This recipe compiles all Python code already at buildout-time, not at run-time.
 You will see right away when there is some illegal syntax.
 
```
Same here, do we link to the package already somewhere else?

stevepiercy replied:
See line 25, the first time it is mentioned. Easier to maintain one instance of hyperlinks than multiple, so go with the first.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -209,74 +217,77 @@ Precompile will be run every time you run buildout. If you want to avoid running
 tells you that there is a class that is using an ``implements`` statement which needs to be replaced by the ``@implementer`` decorator.
 
-Example, this kind of code:
+For example, this kind of code:
```
I understand what the author is trying to say, but from a user point of view, using the wording "this kind of code" is confusing.
We should try to be explicit where possible or re-word it.

stevepiercy replied:
This is the start of a sentence

which is finished on the next narrative line.

This is a common pattern in documentation.

svx replied:
What I mean is not the "for example" I mean "kind of code"
What "kind of code"?
"kind of code" is somewhat reallly really general.
If you are not deep into that, this is maybe not really helpful.

stevepiercy replied:
Reworded in b5e5ac8

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -209,74 +217,77 @@ Precompile will be run every time you run buildout. If you want to avoid running
 
-Relative imports like `import permissions` are no longer permitted. Use `from collective.package import permissions` or `from . import permissions` (not recommended)
+Relative imports like ``import permissions`` are no longer permitted.
+Use ``from collective.package import permissions`` or ``from . import permissions`` (not recommended).
```
Why did we mention a not recommended way?

stevepiercy replied:
Agreed, it should be removed, but I'm not 100% sure whether only the latter of the two or both import styles should be removed. It's not clear.

svx replied:
@plone/framework-team can you shed some light here? Thanks!

jensens replied:
We prefer to not use the dot b/c the full path increases readability. Not always you exactly know where you're standing while reading code, like after a search or at github.

stevepiercy replied:
Fixed in 8278c92.

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -209,74 +217,77 @@ Precompile will be run every time you run buildout. If you want to avoid running
 
-For this step you should have ``Products.PDBDebugMode`` installed. It will make fixing any issues much easier.
+For this step you should have ``Products.PDBDebugMode`` installed.
+It will make fixing any issues much easier.
```
Personally, I would not use "much easier" this can be misleading.

Nothing is 'easy' what is 'easy' for you may not 'easy' for others.
You could change that to maybe effortless or something else?

stevepiercy replied:
Reworded in 4d5ab52

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -286,13 +297,21 @@ For this step you should have ``Products.PDBDebugMode`` installed. It will make
     $ ./bin/test --all -s collective.package
 
-Remember that you can run ``./bin/test -s collective.package -D`` to find yourself in a pdb when a error happens.
+Remember that you can run ``./bin/test -s collective.package -D`` to enter a pdb session when an error occurs.
```
Should we use ``pdb`` ?

stevepiercy replied:
`¯\_(ツ)_/¯`

svx replied:
Stupid GitHub UX ... 
What I meant was to put pdb into `  to make clear that it is a command/program

stevepiercy replied:
LOL. OK, got it fixed in ab5b22b

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -286,13 +297,21 @@ For this step you should have ``Products.PDBDebugMode`` installed. It will make
+*   Choose the link "Build with parameters" in the left menu on the left-hand side.
+*   Fill the fields "ADDON_URL" and "ADDON_BRANCH" with your repository's URL and the branch name ("python3" if you followed these instructions).
+*   Start the build with the "Build" button.
 
```
Maybe we should add a picture?
This is nice because it gives the user a visual and also it makes reading the docs a better experience.

svx replied:
One other thing, we need to add the "gui label" directive to highlit the buttons, this is how we do it in other parts of the docs, too.

stevepiercy replied:
I think this audience—developers upgrading an add-on—can get by without a screenshot. I also do not wish to add more work of maintaining screenshots of UIs, especially of third party applications.

stevepiercy replied:
`:guilabel:` also needs to be added to the [rst Style Guide](https://docs.plone.org/about/contributing/rst-styleguide.html).

stevepiercy replied:
Should guilabel be added to all UI elements? Form controls, links, menu items, etc.?

For now added for just the button in 96a794b

svx replied:
Good question, till now if I remember right, we use it for "actions" like buttons and other stuff the user has to "click" on.
There is no "rule" yet afaik.

stevepiercy replied:
I found lots of guilabels for all UI stuff.

Updated in ccff345

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -324,24 +343,36 @@ An example for a tox-setup can be found in https://github.com/collective/collect
-This is by far the biggest issue when porting to Python 3. Read https://portingguide.readthedocs.io/en/latest/strings.html to be prepared.
+This is by far the biggest issue when porting to Python 3.
+Read https://portingguide.readthedocs.io/en/latest/strings.html to be prepared.
 
```
"Read the portingguide<LINK>" gives the user a better experience

stevepiercy replied:
Added label to link in bbfd8f3

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -324,24 +343,36 @@ An example for a tox-setup can be found in https://github.com/collective/collect
+Read https://portingguide.readthedocs.io/en/latest/strings.html to be prepared.
 
-As a rule of thumb you can assume that in Python 3 everything should be text and only in very rare cases you need to handle bytes.
+As a rule of thumb, you can assume that in Python 3 everything should be text.
```
I would use a "note" directive for the rule of thumb.
This creates a better experience for the user and also makes more clear that it is a "rule of thumb"

stevepiercy replied:
Fixed in 0fac7f2

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -324,24 +343,36 @@ An example for a tox-setup can be found in https://github.com/collective/collect
 
-As a rule of thumb you can assume that in Python 3 everything should be text and only in very rare cases you need to handle bytes.
+As a rule of thumb, you can assume that in Python 3 everything should be text.
+Only in very rare cases you need to handle bytes.
```
This is "tricky" wording.
This may 150% technically correct, but in terms or wording, this is tricky.
I would "just" use "in rare cases"

This is IMHO a bit better because by doing so we do not assume knowledge or something else.
If you are not deep into Plone you do not know if that is a rare or a very rare case.

stevepiercy replied:
The grammar was fixed in 0fac7f2

stevepiercy committed:
Remove "However, "

stevepiercy committed:
Rewrite phrase for clarity

svx commented on manage/upgrading/version_specific_migration/upgrade_to_python3.rst:
```diff
@@ -241,7 +241,7 @@ This kind of error message:
 tells you that there is a class that is using an ``implements`` statement which needs to be replaced by the ``@implementer`` decorator.
 
-For example, this kind of code:
+For example, code that is written as follows:
```
Super awesome! Thanks a lot!!!!

stevepiercy committed:
Add links to first instances of plone.reload and Products.PDBDebugMode

stevepiercy committed:
Reword sentences for Products.PDBDebugMode

stevepiercy committed:
Add label to link for porting guide

stevepiercy committed:
Convert paragraph to note for text/bytes on Python 3; fix grammar.

stevepiercy committed:
Add :guilabel: for UI button

stevepiercy committed:
double backtick pdb

stevepiercy committed:
Add link for source checkout

stevepiercy committed:
Add guilabels
- Use actual <title> of page for Jenkins
- Add missing GitHub auth step

stevepiercy committed:
Improve wording slightly. Still klunky. meh.

stevepiercy committed:
Add warning about pre-release/final release of 5.2

stevepiercy commented:
Thanks for the feedback @svx.

With that, there are two things remaining to be resolved for this PR:

- [ ] https://github.com/plone/documentation/pull/1050#pullrequestreview-209854473
- [x] https://github.com/plone/documentation/pull/1050#pullrequestreview-209854789

svx commented:
@stevepiercy yes!
Let's hope we will get some answers soon, would be nice to get it merged!
Thanks for all your work, time and effort!

stevepiercy committed:
Use fully qualified paths

stevepiercy commented:
@pbauer can you clarify https://github.com/plone/documentation/pull/1050#pullrequestreview-209854473? According to [git blame](https://github.com/plone/documentation/blame/b4c7a02c2a7ff3d2a339634c9afbe7ad4c5f2903/manage/upgrading/version_specific_migration/upgrade_to_python3.rst#L92), you made the change, but we can't figure out what exactly you were trying to convey. Thank you!

pbauer committed:
I released Products.PDBDebugMode so there is no need for a checkout any more

pbauer commented:
Done. Thanks for improving!
